File: .sdk-repos/version-8.9/orchestration-cluster-api-csharp/examples/Authorization.cs
// Compilable usage examples for authorization operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class AuthorizationExamples
{
    #region CreateAuthorization
    // <CreateAuthorization>
    public static async Task CreateAuthorizationExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.CreateAuthorizationAsync(new AuthorizationPropertyBasedRequest
        {
            ResourceType = ResourceTypeEnum.PROCESSDEFINITION,
            PermissionTypes = new List<PermissionTypeEnum> { PermissionTypeEnum.READ, PermissionTypeEnum.UPDATE },
            ResourcePropertyName = "my-process",
            OwnerType = OwnerTypeEnum.USER,
            OwnerId = "user@example.com",
        });

        Console.WriteLine($"Authorization key: {result.AuthorizationKey}");
    }
    // </CreateAuthorization>
    #endregion CreateAuthorization

    #region GetAuthorization

    // <GetAuthorization>
    public static async Task GetAuthorizationExample(AuthorizationKey authorizationKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetAuthorizationAsync(
            authorizationKey);

        Console.WriteLine($"Resource type: {result.ResourceType}");
    }
    // </GetAuthorization>
    #endregion GetAuthorization

    #region SearchAuthorizations

    // <SearchAuthorizations>
    public static async Task SearchAuthorizationsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchAuthorizationsAsync(
            new AuthorizationSearchQuery());

        foreach (var auth in result.Items)
        {
            Console.WriteLine($"Authorization: {auth.AuthorizationKey}");
        }
    }
    // </SearchAuthorizations>
    #endregion SearchAuthorizations

    #region UpdateAuthorization

    // <UpdateAuthorization>
    public static async Task UpdateAuthorizationExample(AuthorizationKey authorizationKey)
    {
        using var client = CamundaClient.Create();

        await client.UpdateAuthorizationAsync(
            authorizationKey,
            new AuthorizationPropertyBasedRequest
            {
                ResourceType = ResourceTypeEnum.PROCESSDEFINITION,
                PermissionTypes = new List<PermissionTypeEnum> { PermissionTypeEnum.READ, PermissionTypeEnum.UPDATE, PermissionTypeEnum.DELETE },
                ResourcePropertyName = "my-process",
                OwnerType = OwnerTypeEnum.USER,
                OwnerId = "user@example.com",
            });
    }
    // </UpdateAuthorization>
    #endregion UpdateAuthorization

    #region DeleteAuthorization

    // <DeleteAuthorization>
    public static async Task DeleteAuthorizationExample(AuthorizationKey authorizationKey)
    {
        using var client = CamundaClient.Create();

        await client.DeleteAuthorizationAsync(authorizationKey);
    }
    // </DeleteAuthorization>
    #endregion DeleteAuthorization

    #region GetAuthentication

    // <GetAuthentication>
    public static async Task GetAuthenticationExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.GetAuthenticationAsync();
        Console.WriteLine($"Authenticated user: {result.Username}");
    }
    // </GetAuthentication>
    #endregion GetAuthentication
}

File: .sdk-repos/version-8.9/orchestration-cluster-api-python/examples/authorization.py
# Compilable usage examples for authorization operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    AuthorizationIdBasedRequest,
    AuthorizationIdBasedRequestPermissionTypesItem,
    AuthorizationIdBasedRequestResourceType,
    AuthorizationKey,
    AuthorizationSearchQuery,
    CamundaClient,
    OwnerTypeEnum,
    Unset,
)


# region CreateAuthorization
def create_authorization_example() -> None:
    client = CamundaClient()

    result = client.create_authorization(
        data=AuthorizationIdBasedRequest(
            resource_type=AuthorizationIdBasedRequestResourceType.PROCESS_DEFINITION,
            permission_types=[
                AuthorizationIdBasedRequestPermissionTypesItem.READ,
                AuthorizationIdBasedRequestPermissionTypesItem.UPDATE,
            ],
            resource_id="my-process",
            owner_type=OwnerTypeEnum.USER,
            owner_id="user@example.com",
        ),
    )

    print(f"Authorization key: {result.authorization_key}")
# endregion CreateAuthorization


# region GetAuthorization
def get_authorization_example(authorization_key: AuthorizationKey) -> None:
    client = CamundaClient()

    result = client.get_authorization(
        authorization_key=authorization_key,
    )

    print(f"Resource type: {result.resource_type}")
# endregion GetAuthorization


# region SearchAuthorizations
def search_authorizations_example() -> None:
    client = CamundaClient()

    result = client.search_authorizations(
        data=AuthorizationSearchQuery(),
    )

    if not isinstance(result.items, Unset):
        for auth in result.items:
            print(f"Authorization: {auth.authorization_key}")
# endregion SearchAuthorizations


# region UpdateAuthorization
def update_authorization_example(authorization_key: AuthorizationKey) -> None:
    client = CamundaClient()

    client.update_authorization(
        authorization_key=authorization_key,
        data=AuthorizationIdBasedRequest(
            resource_type=AuthorizationIdBasedRequestResourceType.PROCESS_DEFINITION,
            permission_types=[
                AuthorizationIdBasedRequestPermissionTypesItem.READ,
                AuthorizationIdBasedRequestPermissionTypesItem.UPDATE,
                AuthorizationIdBasedRequestPermissionTypesItem.DELETE,
            ],
            resource_id="my-process",
            owner_type=OwnerTypeEnum.USER,
            owner_id="user@example.com",
        ),
    )
# endregion UpdateAuthorization


# region DeleteAuthorization
def delete_authorization_example(authorization_key: AuthorizationKey) -> None:
    client = CamundaClient()

    client.delete_authorization(
        authorization_key=authorization_key,
    )
# endregion DeleteAuthorization
